pkg/retry/retry.go

```diff
@@ -221,3 +219,25 @@ func (s *Server) handleRetry() http.Handler {
 		fmt.Fprint(w, http.StatusText(http.StatusAccepted))
 	})
 }
+
+// writeMostRecentCheckpoint is a helper function to write to the checkpoint
+// table with the last successfully processed checkpoint denoted by
+// newCheckpoint.
+func (s *Server) writeMostRecentCheckpoint(ctx context.Context, w http.ResponseWriter,
```

Comment:
Nit: pull the logger from the context instead.

Reply:
ack done